### PR TITLE
Add datacenter flag to controller deployment

### PR DESCRIPTION
**Pull request by thisisnotashwin**
This is a companion PR to https://github.com/hashicorp/consul-k8s/pull/343 which passes the datacenter flag to the controller.

### CODE REVIEW

File: templates/controller-deployment.yaml [hunk 1]
```diff
@@ -58,6 +58,7 @@ spec:
         - |
           consul-k8s controller \
             --webhook-tls-cert-dir=/tmp/controller-webhook/certs \
+            --datacenter={{ .Values.global.datacenter }}
```
Comment (lkysow) [inline templates/controller-deployment.yaml:61]: does this take single `-`'s?

**Review by lkysow — APPROVED** [review]

File: templates/controller-deployment.yaml (hunk 1 above)
Comment (ashwin-venkatesh) [reply templates/controller-deployment.yaml:61]: I think they do. Want me to update all flags with a single `-`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -140,7 +140,8 @@ jobs:
               -kubeconfig="$primary_kubeconfig" \
               -secondary-kubeconfig="$secondary_kubeconfig" \
               -debug-directory="$TEST_RESULTS/debug" \
-              -consul-k8s-image=hashicorpdev/consul-k8s:crd-controller-base-latest
+              -consul-k8s-image=ashwinvenkatesh/consul-k8s:meta@sha256:531a8865c4c585b53962bef8ff7e7df375741b16ee4f012d53f348deb30ba5f9 \
+              -consul-image=ashwinvenkatesh/consul:meta@sha256:ba215ab440990e19e87f5fa0952f981dcbf2605175b97899dfec259ee1abeca5
```
Comment (ashwin-venkatesh) [inline .circleci/config.yml:144]: These will be reset before the code gets merged

File: templates/controller-deployment.yaml (hunk 1 above)
Comment (kschoche) [reply templates/controller-deployment.yaml:61]: I think that makes sense to do at some point

Comment (ashwin-venkatesh) [reply templates/controller-deployment.yaml]: how about now? 😉

**Review by kschoche — APPROVED** [review]
💥 🧨 Looks great!

**Comment by lkysow** [comment]
@ashwin-venkatesh we need this merged I think for the tests to pass now cuz we've made that flag mandatory.

**Pull request merged**